### ISSUE TITLE
Production embroider build – empty Components tab

### ISSUE THREAD

**Issue by gilest** [emberjs/ember-inspector#2125]
**Describe the bug**

When inspecting a production app built with embroider the Components tab is empty.

**To Reproduce**

[Minimal reproduction repo here](https://github.com/gilest/embroider-component-inspector-repro)

- `git clone git@github.com:gilest/embroider-component-inspector-repro.git`
- `yarn`
- `ember s -e production`
- Open ember inspector, visit Components tab
- `<Example />` component is visible ✅ 
- Kill server
- `git checkout embroider`
- `yarn`
- `ember s -e production`
- Open ember inspector, visit Components tab
- Components tab is empty ❌ 

**Expected behavior**

`<Example />` component should be inspectable from the Components tab.

**Screenshots**

Branch: **master**
![Screen Shot 2022-08-31 at 10 49 23 AM](https://user-images.githubusercontent.com/36919/187559386-b65a8302-eb42-4c66-9af0-3201e2c027fc.png)

Branch: **embroider**
![Screen Shot 2022-08-31 at 10 50 51 AM](https://user-images.githubusercontent.com/36919/187559400-213dfeef-371a-4c2d-b7f6-d94afa7f91b3.png)

**Environment**

MacOS 12.5
Chrome 105.0.5195.52

Reproduced on latest ember-cli blueprint

ember-cli@~4.6.0
ember-source@~4.6.0

**Additional context**

May be related to #1745 (although that issue can maybe be closed now as the reported outright crash does not occur).

**Comment by NullVoxPopuli** [comment]
I confirm your reproduction steps behave as you say they do.

I think this may be something specific to your app/settings though. 

I have an app running locally in production mode as well, and the inspector works:
![image](https://user-images.githubusercontent.com/199018/187569847-68c09f46-7b86-4a22-aa32-7935fa6bb3f9.png)

So... this is good, we have an app that works and an app that doesn't, both with embroider -- so ... _something_ is different between them.

**Comment by gilest** [comment]
> I think this may be something specific to your app/settings though.

Sure, but that app is generated by the default blueprint, so that specific thing is the default for most users..

**Comment by gilest** [comment]
Thanks @NullVoxPopuli 

I did some tests and found that a single line from you `ember-cli-build` is required to enable the component inspector to work in my reproduction.

```js
  app.import('vendor/ember/ember-template-compiler.js');
```

I've pushed this to the `embroider-working` branch of the reproduction repo.

**Comment by rwwagner90** [comment]
@NullVoxPopuli is this something people should be adding in manually?

**Comment by gilest** [comment]
Adding that line to `ember-cli-build` does not resolve this issue in a few ember `3.28.8` apps I tried it in.

**Comment by NullVoxPopuli** [comment]
>  is this something people should be adding in manually?

it should not, adding the template-compiler like this is (imo) "intimate API" for run-time compiled components, which 99.999% of apps are not doing 🙃

**Comment by gilest** [comment]
@NullVoxPopuli do you have any insight about the best repo for this issue to reside within?

**Comment by bendemboski** [comment]
I am also seeing this in an app running:
* embroider 1.8.0 with static<everything>: true
* ember-cli 4.7
* ember-source 4.4.2

**Comment by rwwagner90** [comment]
@chancancode any ideas here?

**Comment by patricklx** [comment]
somehow _DEBUG_RENDER_TREE is false https://github.com/emberjs/ember-inspector/blob/main/ember_debug/libs/capture-render-tree.js#L10
although it should be set to true by ember-inspector: https://github.com/emberjs/ember-inspector/blob/main/skeletons/web-extension/boot.js#L13

**Comment by patricklx** [comment]
it probably is getting set. BUT:
classic build extends the existing EmberENV :
```
window.EmberENV = function(e, t) {
    for (var r in t)
        e[r] = t[r]
    return e
}(window.EmberENV || {}, {
    FEATURES: {},
    EXTEND_PROTOTYPES: {
        Date: !1
    },
    _APPLICATION_TEMPLATE_WRAPPER: !1,
    _DEFAULT_ASYNC_OBSERVERS: !0,
    _JQUERY_INTEGRATION: !1,
    _TEMPLATE_ONLY_GLIMMER_COMPONENTS: !0
})
```

whereas embroid build overwrites:
 ```
 window.EmberENV = {
    FEATURES: {},
    _APPLICATION_TEMPLATE_WRAPPER: !1,
    _DEFAULT_ASYNC_OBSERVERS: !0,
    _JQUERY_INTEGRATION: !1,
    _TEMPLATE_ONLY_GLIMMER_COMPONENTS: !0
};
 ```

**Comment by gilest** [comment]
Can confirm that this fixes the issue[ in the minimal reproduction app](https://github.com/gilest/embroider-component-inspector-repro/commit/9ff5fdbfe6d3159f76d450d781879f444e95b5be).

**Comment by patricklx** [comment]
@NullVoxPopuli in limber I see some duplicate modules in the source for "@ember/-internals/environment/index". One without glimmer DEBUG enabled and another with DEBUG enabled... the one with DEBUG enabled is processed and thus sets _DEBUG_RENDER_TREE  to true

**Comment by NullVoxPopuli** [comment]
Hmm, duplicate modules is suspicious.
Gonna look in to that

**Comment by patricklx** [comment]
ah, thats the ember-template-compiler... its build with debug enabled..

**Comment by NullVoxPopuli** [comment]
Ah, right. Then my issue (accidental working) will be fixed by the template layering RFC

**Comment by rwwagner90** [comment]
Is there a good way to fix this in inspector? I don't know much about how embroider works.

**Comment by gilest** [comment]
@rwwagner90 Should be fixed once this PR is merged and released: embroider-build/embroider/pull/1252

This issue can probably be closed

**Comment by patricklx** [comment]
the pr is merged and this can be closed. Although it's not released yet
